server/src/com/thoughtworks/go/server/service/GoConfigService.java

```diff
@@ -929,6 +930,15 @@ public String unmodifiedMd5() {
         });
     }
 
```

Comment:
Would prefer if we move both these methods to `EnvironmentConfigService` instead. From there, you could invoke `public ConfigSaveState updateConfig(UpdateConfigCommand command)` (which internally invokes `goConfigDao.updateConfig`)